gravitee-apim-console-webui/src/management/application/details/user-group-access/transfer-ownership/application-general-transfer-ownership.component.ts

```diff
@@ -13,11 +13,123 @@
+
+  mode: 'USER' | 'GROUP' | 'HYBRID';
+  form: UntypedFormGroup;
+  applicationMembers: Member[];
+  roles: Role[];
+  groups: Group[];
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly roleService: RoleService,
+    private readonly applicationMembersService: ApplicationMembersService,
+    private readonly applicationTransferOwnershipService: ApplicationTransferOwnershipService,
+    private readonly matDialog: MatDialog,
+    private readonly snackBarService: SnackBarService,
+    private readonly groupService: GroupService,
+  ) {}
+
+  ngOnInit(): void {
+    combineLatest([
+      this.applicationMembersService.get(this.activatedRoute.snapshot.params.applicationId),
+      this.roleService.list('APPLICATION'),
+      this.groupService.list(),
+    ])
+      .pipe(
+        tap(([members, roles, groupList]) => {
+          this.groups = groupList;
+          this.applicationMembers = members.filter((member) => member.role !== 'PRIMARY_OWNER');
+          this.roles = roles.filter((role) => role.name !== 'PRIMARY_OWNER');
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => {
+        const defaultRole = this.roles.find((role) => role.default);
+        this.initForm(defaultRole);
+      });
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.complete();
+  }
+
+  onSubmit() {
+    const newRole = this.form.get('roleId').value;
+
+    const confirmDialog = this.matDialog.open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
+      data: {
+        title: 'Transfer Application ownership',
+        content: `This action cannot be undone. If you are the primary owner of this Application, your role will be set to <code>${newRole}</code>.`,
+        confirmButton: 'Transfer',
+      },
+      role: 'alertdialog',
+      id: 'confirmTransferDialog',
+    });
+
+    const userMode = this.form.get('userOrGroup').value;
+    const user = this.form.get('user').value;
+    const isUserMode = userMode === 'user' || userMode === 'applicationMember';
+
+    const transferOwnershipToUser: any = {
+      id: user?.id,
+      reference: user?.reference,
+      role: newRole,
+    };
+
+    confirmDialog
+      .afterClosed()
+      .pipe(
+        filter((confirmed) => confirmed),
+        switchMap(() =>
+          this.applicationTransferOwnershipService.transferOwnership(
+            this.activatedRoute.snapshot.params.applicationId,
+            isUserMode ? transferOwnershipToUser : {},
+          ),
+        ),
+        tap(
+          () => this.snackBarService.success('Transfer ownership done.'),
+          ({ error }) => {
+            this.snackBarService.error(error.message);
+            return EMPTY;
+          },
+        ),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.ngOnInit);
+  }
+
+  private initForm(defaultRole: any) {
+    this.form = new UntypedFormGroup({
```

Comment:
Note required. But maybe we can star to Type ou forms 🤷‍♂️

Reply:
Done